src/main/java/net/dv8tion/jda/api/entities/subscription/SubscriptionImpl.java

```diff
@@ -0,0 +1,114 @@
+package net.dv8tion.jda.api.entities.subscription;
+
+import net.dv8tion.jda.internal.utils.EntityString;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.time.OffsetDateTime;
+import java.util.List;
+
+
+public class SubscriptionImpl implements Subscription
+{
+    private final long id;
+    private final long subscriberId;
+    private final List<Long> skuIDs;
+    private final List<Long> entitlementIDs;
+    private final List<Long> renewalSkuIDs;
+    private final OffsetDateTime currentPeriodStart;
+    private final OffsetDateTime currentPeriodEnd;
+    private final OffsetDateTime canceledAt;
+    private final SubscriptionStatus status;
+
+    public SubscriptionImpl(final long id, final long subscriberId, @Nonnull final List<Long> skuIDs,
+                            @Nonnull final List<Long> entitlementIDs, @Nullable final List<Long> renewalSkuIDs,
+                            @Nonnull final OffsetDateTime currentPeriodStart, @Nonnull final OffsetDateTime currentPeriodEnd, @Nullable final OffsetDateTime canceledAt,
+                            @Nonnull final SubscriptionStatus status)
+    {
+
+        this.id = id;
+        this.subscriberId = subscriberId;
+        this.skuIDs = skuIDs;
+        this.entitlementIDs = entitlementIDs;
+        this.renewalSkuIDs = renewalSkuIDs;
+        this.currentPeriodStart = currentPeriodStart;
+        this.currentPeriodEnd = currentPeriodEnd;
+        this.canceledAt = canceledAt;
+        this.status = status;
+    }
+
+    @Nonnull
+    @Override
+    public long getIdLong()
+    {
+        return id;
+    }
+
+    public long getSubscriberIdLong()
+    {
+        return subscriberId;
+    }
+
+    public @NotNull List<Long> getSkuIdsLong()
```

Comment:
1. Use `@Nonnull` instead
2. Annotations are on the line before the method declaration